spyder/api/widgets.py

```diff
@@ -0,0 +1,1167 @@
+            for item in items:
+                if isinstance(item, QAction):
+                    self.removeAction(item)
+                else:
+                    action = [a for a in self.actions()
+                              if self.widgetForAction(a) == item]
+                    if action:
+                        self.removeAction(action[0])
+
+                sec_items.append([sec, item])
+
+            sep = QAction(self)
+            sep.setSeparator(True)
+            sec_items.append((None, sep))
+
+        sec_items.pop()
+
+        # Remove separator actions
+        for act in self.actions():
+            if act.isSeparator():
+                self.removeAction(act)
+
+        if self._corner_widget is not None:
+            for (sec, item) in sec_items:
+                if isinstance(item, QAction):
+                    insert_method = super().insertAction
+                    add_method = super().addAction
+                else:
+                    insert_method = super().insertWidget
+                    add_method = super().addWidget
+
+                if before is None:
+                    insert_method(self._corner_separator, item)
+                elif item == action_or_widget:
+                    insert_method(before, item)
+                else:
+                    insert_method(self._corner_separator, item)
+        else:
+            if isinstance(action_or_widget, QAction):
+                insert_method = super().insertAction
+                add_method = super().addAction
+            else:
+                insert_method = super().insertWidget
+                add_method = super().addWidget
+
+            if before is None:
+                add_method(action_or_widget)
+            else:
+                insert_method(before, item)
+
+    def _set_corner_widget(self, corner_widget):
+        """
+        Add the given corner widget to this toolbar.
+
+        A stretcher widget is added before the corner widget so that
+        its position is forced to the right side of the toolbar when the
+        toolbar is resized.
+        """
+        self._corner_widget = corner_widget
+        if corner_widget is not None:
+            stretcher = create_toolbar_stretcher()
+            self._corner_separator = super().addWidget(stretcher)
+            super().addWidget(self._corner_widget)
+        else:
+            self._corner_separator = None
+
+    def _setup_style(self):
+        """
+        Set the style of this toolbar with a stylesheet.
+        """
+        if is_dark_interface():
+            self.setStyleSheet(
+                "QToolButton {background-color: transparent;} "
+                "QToolButton:!hover:!pressed {border-color: transparent} "
+                "QToolBar {border: 0px;  background: rgb(25, 35, 45);}"
+            )
+        else:
+            self.setStyleSheet("QToolBar {border: 0px;}")
+
+
+class SpyderWidget(QMainWindow, SpyderWidgetMixin):
```

Comment:
Can we have a different name? This name could refer to any widget in Spyder ... I think this was discussed before, but if not, one proposal is `PluginMainWidget`

Reply:
Yes, makes sense!